### PR TITLE
OCPCLOUD-2430,CFE-684: Rebase k8s 1.29 and update dependencies

### PR DESCRIPTION
**Pull request by RadekManak**
Updates k8s to 1.29 and other dependencies to latest version.

Most breaking changes were because of controller-runtime and govmomi.

### CODE REVIEW

File: cmd/vsphere/main.go [hunk 1]
```diff
@@ -189,14 +189,14 @@ func main() {
...
+		klog.Fatalf("unable to add ipamv1alpha1 to scheme: %v", err)
 	}
 
 	if err := capimachine.AddWithActuator(mgr, machineActuator); err != nil {
 		klog.Fatal(err)
 	}
 
-	ctrl.SetLogger(klogr.New())
+	ctrl.SetLogger(textlogger.NewLogger(textlogger.NewConfig()))
 	setupLog := ctrl.Log.WithName("setup")
```
[the remark this answers is not in the document]
Comment (RadekManak) [reply cmd/vsphere/main.go:202]: Is it correct like this?

[the remark this answers is not in the document]
Comment (JoelSpeed) [reply cmd/vsphere/main.go:202]: Yes, but line 200 needs to replace line 46 where we klog.InitFlags, that's not rquired anymore

**Comment by RadekManak** [comment]
/hold
VSphere has issues.

**Comment by bharath-b-rh** [comment]
@RadekManak Could you please pull the latest version of `openshift/api`, https://github.com/openshift/api/pull/1506 got merged yesterday. Thank you!

**Comment by JoelSpeed** [comment]
Machines/MachineSets all look healthy in the vSphere jobs even though they've failed, @RadekManak should we let this one merge?

Otherwise LGTM so will tag

/approve
/lgtm

**Pull request merged**